app/models/miq_server/worker_management/kubernetes.rb

```diff
@@ -14,15 +14,28 @@ def sync_monitor
   end
 
   def sync_from_system
+    # All miq_server instances have to reside on the same Kubernetes cluster, so
+    # we only have to sync the list of pods and deployments once
+    return unless my_server.is_master?
+
     ensure_kube_monitors_started
-    cleanup_orphaned_worker_rows
-    sync_deployment_settings
```

Comment:
Moved these since they weren't related to syncing with the underlying system, they only use the result of this sync.